src/main/java/rx/internal/operators/OperatorEagerConcatMap.java

```diff
@@ -0,0 +1,299 @@
+/**
+ * Copyright 2014 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package rx.internal.operators;
+
+import java.util.*;
+import java.util.concurrent.atomic.*;
+
+import rx.*;
+import rx.Observable;
+import rx.Observable.Operator;
+import rx.exceptions.Exceptions;
+import rx.functions.*;
+import rx.internal.util.atomic.SpscLinkedArrayQueue;
+import rx.subscriptions.Subscriptions;
+
+public final class OperatorEagerConcatMap<T, R> implements Operator<R, T> {
+    final Func1<? super T, ? extends Observable<? extends R>> mapper;
+    final int bufferSize;
+    public OperatorEagerConcatMap(Func1<? super T, ? extends Observable<? extends R>> mapper, int bufferSize) {
+        this.mapper = mapper;
+        this.bufferSize = bufferSize;
+    }
+    
+    @Override
+    public Subscriber<? super T> call(Subscriber<? super R> t) {
+        EagerOuterSubscriber<T, R> outer = new EagerOuterSubscriber<T, R>(mapper, bufferSize, t);
+        outer.init();
+        return outer;
+    }
+    
+    static final class EagerOuterProducer extends AtomicLong implements Producer {
+        /** */
+        private static final long serialVersionUID = -657299606803478389L;
+        
+        final EagerOuterSubscriber<?, ?> parent;
+        
+        public EagerOuterProducer(EagerOuterSubscriber<?, ?> parent) {
```

Comment:
remove `public`? (All the constructors in this file for non-public static classes are marked as public).